### PR TITLE
Added public IP label to Azure SD targets.

### LINKED ISSUE #3970: azure sd with public IP
**What did you do?**

trying to scrape azure instances with their public IP

**What did you expect to see?**
metrics

**What did you see instead? Under which circumstances?**
the current azure implementation https://github.com/prometheus/prometheus/blob/master/discovery/azure/azure.go#L38 does not allow for Public Azure IP's to be scraped thus every azure location/ ressource group would need their own prometheus instances 

* Prometheus version:
any

### PR DESCRIPTION
**Pull request by h7kanna**
@brian-brazil 

Hi,
I added Public IP information to Azure SD labels.
I needed this to do some load tests on Public Azure VMs.
Resolves #3970

### CODE REVIEW

File: discovery/azure/azure.go [hunk 1]
```diff
@@ -291,6 +297,27 @@ func (d *Discovery) refresh() (tg *targetgroup.Group, err error) {
 							labels[azureLabelMachinePrivateIP] = model.LabelValue(*ip.Properties.PrivateIPAddress)
 							address := net.JoinHostPort(*ip.Properties.PrivateIPAddress, fmt.Sprintf("%d", d.port))
 							labels[model.AddressLabel] = model.LabelValue(address)
+							// Check if Public IP is also present
```
Comment (brian-brazil) [inline discovery/azure/azure.go]: Full stop

File: discovery/azure/azure.go [hunk 2]
```diff
@@ -291,6 +297,27 @@ func (d *Discovery) refresh() (tg *targetgroup.Group, err error) {
...
+							// Get the Public IP address information via separate call to the network provider.
```
Comment (brian-brazil) [inline discovery/azure/azure.go]: Azure has a quite low rate limit for API requests, so an additional request per instance let alone per network interface is problematic. Is there a way to do this without any additional requests?

Comment (h7kanna) [reply discovery/azure/azure.go]: Unfortunately there is no other way as public IP is a separate resource in Azure RM API and only referenced by ID in the network interface.

Comment (brian-brazil) [reply discovery/azure/azure.go]: That's unfortunate, this isn't the first PR we've had to reject for this. Hopefully Azure will have higher API limits or more denormalised APIs in the future.

File: discovery/azure/azure.go (hunk 1 above)
Comment (h7kanna) [reply discovery/azure/azure.go]: Done

File: discovery/azure/azure.go (hunk 2 above)
Comment (h7kanna) [reply discovery/azure/azure.go]: Yes I understand. But I was thinking this network call will be triggered only for VMs which have a public IP. Most VMs should have only Private IP(internal). 
Thanks for quick review. Will close this.

**Pull request closed without merging**